### ISSUE TITLE
On a YCM-based superbuild, if there is no BuildPackage.cmake file, the superbuild fails silently

### ISSUE THREAD

**Issue by traversaro** [robotology/ycm#126]
cc @jeljaik

**Comment by drdanz** [comment]
If I understand correctly if you have a `find_or_build_package(Foo)` call and there is no `BuildFoo.cmake` file, it fails silently, but what is failing exactly? Just the package is not built or CMake fails?

**Comment by traversaro** [comment]
Sorry: the package is not built.

**Comment by drdanz** [comment]
Ok, so the problem is that the package is not built, but you don't get a feedback, right?

**Comment by traversaro** [comment]
Exactly. Perhaps we should use `find_or_build_package(Foo REQUIRED)` ? I never considered doing that.

**Comment by drdanz** [comment]
I don't remember if that is supported, if it is I might have done this on purpose... :smile:
